File: Examples/CSharp/AsposePDF/Links-Actions/SpecifyPageWhenViewing.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.InteractiveFeatures;

namespace CSharp.AsposePdf.LinksActions
{
    public class SpecifyPageWhenViewing
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_LinksActions();

            // load the PDF file
            Document doc = new Document(dataDir+ "SpecifyPageWhenViewing.pdf");
            // get the instance of second page of document
            Page page2 = doc.Pages[2];
            // create the variable to set the zoom factor of target page
            double zoom = 1;
            // create GoToAction instance
            Aspose.Pdf.InteractiveFeatures.GoToAction action = new Aspose.Pdf.InteractiveFeatures.GoToAction();
            // Go to 2 page
            action.Destination = new XYZExplicitDestination(page2, 0, page2.Rect.Height, zoom);
            // set the document open action
            doc.OpenAction = action;
            // save updated document
            doc.Save(dataDir + "goto2page.pdf");
 
            
            
        }
    }
}

File: Examples/VisualBasic/AsposePDF/Links-Actions/SpecifyPageWhenViewing.vb
Imports System.IO
Imports System
Imports Microsoft.VisualBasic
Imports Aspose.Pdf.Annotations
Imports Aspose.Pdf

Namespace AsposePDF.LinksActions
    Public Class SpecifyPageWhenViewing
        Public Shared Sub Run()
            ' ExStart:SpecifyPageWhenViewing
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_LinksActions()

            ' Load the PDF file
            Dim doc As New Document(dataDir & Convert.ToString("SpecifyPageWhenViewing.pdf"))
            ' Get the instance of second page of document
            Dim page2 As Page = doc.Pages(2)
            ' Create the variable to set the zoom factor of target page
            Dim zoom As Double = 1
            ' Create GoToAction instance
            Dim action As New GoToAction(doc.Pages(2))
            ' Go to 2 page
            action.Destination = New XYZExplicitDestination(page2, 0, page2.Rect.Height, zoom)
            ' Set the document open action
            doc.OpenAction = action
            ' Save updated document
            doc.Save(dataDir & Convert.ToString("goto2page_out_.pdf"))
            ' ExEnd:SpecifyPageWhenViewing                        
        End Sub
    End Class
End Namespace
